File: C#/oddgnome.cs
using System;

public class oddgnome {
    public static void Main (string[] args) {
        int cases = int.Parse (Console.ReadLine ());
        while (cases-- > 0) {
            string[] line = Console.ReadLine ().Split (' ');
            int[] nums = new int[line.Length - 2];
            for (int x = 0; x < nums.Length; x++)
                nums[x] = int.Parse (line[x + 1]);
            for (int x = 0; x < line.Length - 1; x++)
                if (nums[x] != nums[x + 1] - 1) {
                    Console.WriteLine (x + 2);
                    break;
                }
        }
    }
}

File: Java/oddgnome.java

// https://open.kattis.com/problems/oddgnome
import java.util.ArrayList;
import java.util.Scanner;

public class oddgnome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int cases = sc.nextInt();
        sc.nextLine();
        while (cases-- > 0) {
            ArrayList<Integer> gnomes = new ArrayList<>();
            String[] nums = sc.nextLine().split(" ");
            for (int x = 0; x < nums.length; x++)
                gnomes.add(Integer.parseInt(nums[x]));
            gnomes.remove(0);
            gnomes.remove(gnomes.size() - 1);
            for (int x = 0; x < gnomes.size(); x++)
                if (gnomes.get(x + 1) - gnomes.get(x) != 1) {
                    System.out.println(x + 2);
                    break;
                }
        }
        sc.close();
    }
}